clustered/client/src/main/java/org/ehcache/clustered/client/internal/service/DefaultClusteringService.java

```diff
@@ -333,6 +363,51 @@ public void releaseServerStoreProxy(ServerStoreProxy storeProxy) {
     }
   }
 
+  <K extends Serializable, V extends Serializable> ConcurrentMap<K, V> getConcurrentMap(ClusteredCacheIdentifier identifier, String name, Class<K> keyClass, Class<V> valueClass) {
+    Tuple<DefaultClusterCacheIdentifier, ClusteredMapRepository> tuple = knownPersistenceSpaces.get(identifier.getId());
+    if (tuple == null) {
+      throw new AssertionError("Lost a space?? " + identifier);
+    }
+    ClusteredMapRepository mapRepository = tuple.second;
+    while (true) {
+      ConcurrentClusteredMap map = mapRepository.getMap(name);
+      if (map == null) {
+        mapRepository.addNewMap(name, createConcurrentClusteredMap(name, keyClass, valueClass));
+      } else {
+        return map;
+      }
+    }
+  }
+
+  private Tuple<EntityRef<ConcurrentClusteredMap, Object>, ConcurrentClusteredMap> createConcurrentClusteredMap(String name, Class<?> keyClass, Class<?> valueClass) {
+    try {
+      EntityRef<ConcurrentClusteredMap, Object> clusteredMapRef = clusterConnection.getEntityRef(ConcurrentClusteredMap.class, ConcurrentClusteredMap.VERSION, name);
+      ConcurrentClusteredMap clusteredMap;
+      try {
+        clusteredMap = clusteredMapRef.fetchEntity();
+      } catch (EntityNotFoundException e) {
+        clusteredMapRef.create(null);
+        clusteredMap = clusteredMapRef.fetchEntity();
+      }
+      clusteredMap.setTypes(keyClass, valueClass);
```

Comment:
Shouldn't this be inside the exception catch?  Why would we call setTypes on an pre-existing object?